File: data/gpt-4o/readable_format_HPEC/cpp/file034.cpp
#include <stdio.h>
#include <omp.h>

int main() {
    int x = 0, y = 2;
#pragma omp parallel
    {
#pragma omp single
        {
#pragma omp task shared(x)
            {
                x = x + 1;
            }
#pragma omp task shared(y)
            {
                y = y - 1;
            }
#pragma omp taskwait
            printf("x=%d
", x);
#pragma omp taskwait
            printf("y=%d
", y);
        }
    }
    return 0;
}

File: data/StarCoder2/readable_format_HPEC/fortran/file034.f90
program DRB166_taskdep4_orig_no_omp50
    use omp_lib
    implicit none

    !$omp parallel
    !$omp single
    call foo()
    !$omp end single
    !$omp end parallel
contains
    subroutine foo()
        implicit none
        integer :: x, y
        x = 0
        y = 2

        !$omp task depend(inout: x) shared(x)
        x = x+1                                 !!1st Child Task
        !$omp end task

        !$omp task shared(y)
        y = y-1                                 !!2nd child task
        !$omp end task

        !$omp taskwait depend(in: x)            !!1st taskwait

        print*, "x=", x

        !$omp taskwait                          !!2nd taskwait

        print*, "y=", y

    end subroutine
end program
